aiida/orm/nodes/data/singlefile.py

```diff
@@ -71,12 +71,13 @@ def open(self, path=None, mode='r'):
         with self.base.repository.open(path, mode=mode) as handle:
             yield handle
 
-    def get_content(self):
+    def get_content(self, mode='r'):
```

Comment:
```suggestion
    def get_content(self, mode:str = 'r') -> str | bytes:
```
The return type is certain, why not add the type hint?

Reply:
Good point. I have added it. And while I was at it, I have added a separate commit that improves typing in general in this module